### PR TITLE
Initial support for openapiv3

### PR DESCRIPTION
**Pull request by AlisCode**
## Description

Implementation of #106 .
Allows to import Slumber recipes from an OpenAPI specification.

## Known Risks

Not really applicable as this is a new feature, all stuff that is known to not work is clearly commented with a TODO.

Namely, it doesn't look like slumber supports : 
* Cookies
* Path parameters
* OAuth2
* OpenIDConnect

## QA

I added the classic `petstore` sample OpenAPI specification as a test. Feel free to import that and see if that works :)
There is one integration test that tests the feature as a whole, and then a few unit tests for the OpenAPI "reference resolving" module.   

## Checklist

- [x] Have you read `CONTRIBUTING.md` already?
- [x] Did you update `CHANGELOG.md`?
  - Only user-facing changes belong in the changelog. Internal changes such as refactors should only be included if they'll impact users, e.g. via performance improvement.
- [x] Did you remove all TODOs?

-> I think I'd like to discuss a bit more before removing the TODOs and opening more issues, as these are not exactly small features :)

### CODE REVIEW

File: src/collection/openapiv3/mod.rs [hunk 1]
```diff
@@ -0,0 +1,433 @@
...
+        // Load profiles
+        let mut profiles = IndexMap::default();
+        for server in servers {
+            let Server {
+                url,
+                variables,
+                description: _,
+                extensions: _,
+            } = server;
+            let mut data = IndexMap::default();
```
Comment (LucasPickering) [inline src/collection/openapiv3/mod.rs:186]: Same as above, could you please use iterator syntax to generate this instead?

File: src/collection/openapiv3/mod.rs [hunk 2]
```diff
@@ -0,0 +1,433 @@
...
+                    header => {
+                        headers_params
+                            .insert(header.to_string(), Template::empty());
+                    }
+                }
+            }
+            // TODO(path_parameters): Slumber does not support Path parameters
+            Parameter::Path { .. } => {
+                warn!("Unsupported parameter type: Path");
+            }
```
Comment (LucasPickering) [inline src/collection/openapiv3/mod.rs:277]: We could support these by mapping the path param to a template key, e.g. `/pet/{petId}` -> `/pet/{{petId}}`. What do you think? We would then probably want to insert `petId` (with an empty value?) into each profile

**Review by LucasPickering — CHANGES_REQUESTED** [review]
I haven't gone through it entirely, but this should be enough feedback to keep you busy until I have more time to test this.

File: src/collection/openapiv3/mod.rs [hunk 3]
```diff
@@ -0,0 +1,433 @@
...
+        }?;
+        // The following is quoted directly from the specifications of parameter objets
+        // see: https://spec.openapis.org/oas/v3.0.3#parameter-object
+        match param {
+            Parameter::Query { parameter_data, .. } => {
+                // the name corresponds to the parameter name used by the in property.
+                query_params.insert(parameter_data.name, Template::empty());
+            }
+            Parameter::Header { parameter_data, .. } => {
+                // if the name field is "Accept", "Content-Type" or "Authorization", the parameter definition SHALL be ignored.
```
[the remark this answers is not in the document]
Comment (AlisCode) [reply src/collection/openapiv3/mod.rs:261]: This is from the Open API specs : https://spec.openapis.org/oas/v3.0.3#fixed-fields-9

**Comment by LucasPickering** [comment]
@AlisCode Do you intend to keep working on this? If not I can take it over

**Comment by AlisCode** [comment]
Hi @LucasPickering, sorry I lost the focus on this. Feel free to take it from there. I'd be glad to review the code when you're done!

**Review by LucasPickering — APPROVED** [review]

**Comment by LucasPickering** [comment]
@AlisCode Thanks for your work on this! I was pretty slow finishing it up but finally got it done. I'm sure there's lots of gaps in the transformation still but at least we have something to build off of now.

**Pull request merged**